transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpConnector.java

```diff
@@ -140,9 +161,88 @@ protected boolean canConnect(String transportName) {
     }
 
     @Override
-    protected <T extends ConnectFuture> ConnectFuture connectInternal(final ResourceAddress address, final IoHandler handler,
-                                                                      final IoSessionInitializer<T> initializer) {
-        final DefaultConnectFuture bridgeConnectFuture = new DefaultConnectFuture();
+    protected <T extends ConnectFuture> ConnectFuture connectInternal(final ResourceAddress address,
+                final IoHandler handler, final IoSessionInitializer<T> initializer) {
+
+        final ConnectFuture connectFuture =  new DefaultConnectFuture();
+        final ResourceAddress transportAddress = address.getTransport();
+
+        if (transportAddress != null) {
+            Executor ioExecutor = null;
+            boolean ioAligned = isIoAligned();
+            if (!ioAligned) {
+                ioExecutor = getIoExecutor();
+            }
+
+            if (ioAligned || ioExecutor == null) {
```

Comment:
There are two cases:
1) acceptor+proxy+connector. In this case, the executor won't be null. This applies to http proxy 
2) connector. In this case, the executor can be null and will get associated with an executor later.

John want both the cases to work.